misk-crypto/src/main/kotlin/misk/crypto/FieldLevelEncryptionPacket.kt

```diff
@@ -0,0 +1,358 @@
+package misk.crypto
+
+import com.google.common.io.ByteStreams
+import misk.crypto.FieldLevelEncryptionPacket.EntryType
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.io.DataInputStream
+import java.io.DataOutput
+import org.apache.commons.collections4.map.CaseInsensitiveMap
```

Comment:
I think you want `TreeMap(String.CASE_INSENSITIVE_ORDER)` in which case you can avoid introducing a dependency on the Apache commons library

Reply:
(I really, really do not want that dependency)